homeassistant/components/broadlink/light.py

```diff
@@ -0,0 +1,207 @@
+"""Support for Broadlink lights."""
+from abc import ABC, abstractmethod
+import logging
+
+from broadlink.exceptions import BroadlinkException
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP,
+    ATTR_HS_COLOR,
+    COLOR_MODE_COLOR_TEMP,
+    COLOR_MODE_HS,
+    COLOR_MODE_UNKNOWN,
+    LightEntity,
+)
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BROADLINK_COLOR_MODE_RGB = 0
+BROADLINK_COLOR_MODE_WHITE = 1
+BROADLINK_COLOR_MODE_SCENES = 2
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Broadlink light."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+
+    if device.api.type == "LB1":
+        lights = [BroadlinkLB1Light(device)]
```

Comment:
> We put another conditional here. If the data received from the update coordinator contains "hue" and "saturation", we initialize a BroadlinkRGBLight(device). Otherwise, we initialize BroadlinkWhiteLight(device).

Instead of checking for "hue" and "saturation" presence in the data received from the update coordinator, wouldn't it suffice to check if device.api.devtype == 0x60C8 ?

Reply:
No. The first option is universal, it works for all RGB devices, even from different classes. The second is harder to maintain.